### DISCUSSION TITLE
Clarification on intended behaviour

### DISCUSSION THREAD

**Discussion by terryluan12** [zen-fs/core#199] [Q&A]
From #78, I discovered that when reading a file, zen-fs always goes through the `promises.ts:readFile` and subsequently the `file.ts:read` function. 

If you use the Overlay backend, the `file.ts:read` function calls  `overlay.ts:sync` which calls the _writable backend's syc function. If the backends are not identical however, this will result in a `Error: ENOENT: No such file or directory` error. 

I'm working to fix this, but I'm unsure about the intended behaviour. Could I get some guidance on this?

**Comment by james-pre** [comment]
@terryluan12, thanks for the question.

The `Overlay` backend is essentially a [copy-on-write](https://en.wikipedia.org/wiki/Copy-on-write) system for ZenFS. It keeps track of only the changed and deleted files.

**Comment by james-pre** [comment]
@terryluan12 Can I close this issue?

**Comment by terryluan12** [comment]
Yep, thanks for clarifying!
